src/recon_test/test_blocks_on_cylindrical_projectors.cxx

```diff
@@ -871,6 +872,89 @@ BlocksTests::run_intersection_with_cylinder_test()
+      num_axial_pos_per_segment[i] = scanner_sptr->get_num_rings() - abs(i);
+    }
+
+  scanner_sptr->set_scanner_geometry("BlocksOnCylindrical");
+  scanner_sptr->set_up();
+  auto projdata_info_sptr
+      = std::make_shared<ProjDataInfoBlocksOnCylindricalNoArcCorr>(scanner_sptr,
+                                                                   num_axial_pos_per_segment,
+                                                                   min_ring_diff_v,
+                                                                   max_ring_diff_v,
+                                                                   scanner_sptr->get_max_num_views(),
+                                                                   scanner_sptr->get_max_num_non_arccorrected_bins());
+  //  scanner_sptr->set_scanner_geometry("Cylindrical");
+  //  scanner_sptr->set_up();
+  //  auto projdata_info_sptr = std::make_shared<ProjDataInfoCylindricalNoArcCorr>(scanner_sptr,
+  //                                                                               num_axial_pos_per_segment,
+  //                                                                               min_ring_diff_v,
+  //                                                                               max_ring_diff_v,
+  //                                                                               scanner_sptr->get_max_num_views(),
+  //                                                                               scanner_sptr->get_max_num_non_arccorrected_bins());
+
+  auto exam_info_sptr = std::make_shared<ExamInfo>(ImagingModality::PT);
+  auto projdata_sptr = std::make_shared<ProjDataInMemory>(exam_info_sptr, projdata_info_sptr);
+
+  auto origin = CartesianCoordinate3D<float>(0, 0, 0);
+  auto volume_dimensions = CartesianCoordinate3D<int>(-1, -1, -1);
+  auto volume_sptr
+      = std::make_shared<VoxelsOnCartesianGrid<float>>(exam_info_sptr, *projdata_info_sptr, 1, origin, volume_dimensions);
+
+  // Now run the test
+  volume_sptr->fill(0.0);
+  projdata_sptr->fill(1.0);
+
+  back_projector.set_up(projdata_info_sptr, volume_sptr);
+  back_projector.back_project(*volume_sptr, *projdata_sptr, 0, projdata_info_sptr->get_num_views());
+
+  bool test_ok = true;
+  std::ostringstream oss;
+  oss << "BlocksTests::run_back_projection_test_with_axial_buckets: Central voxel values are:" << std::endl;
+
+  auto centre_axial_values = std::vector<float>(volume_sptr->get_z_size());
+  for (int z = volume_sptr->get_min_z(); z <= volume_sptr->get_max_z(); z++)
+    {
+      centre_axial_values[z] = (*volume_sptr)[z][0][0];
+      oss << "\tz = " << z << "/" << volume_sptr->get_max_z() << " is " << centre_axial_values[z] << std::endl;
+      if (test_ok)
+        {
+          test_ok = check(centre_axial_values[z] > 0,
+                          "BlocksTests::run_back_projection_test_with_axial_buckets: Central voxel value <= 0");
+          everything_ok = everything_ok && test_ok;
+        }
+    }
```

Comment:
What other tests can be done to validate the geometries are acceptable? 

Could consider detector coordinate z components monotonically increase. This would require access to a sorted list of them. May require the splitting of `build_crystal_maps` into components (not a bad idea anyway).